### ISSUE TITLE
Loss: 0 after ~15 steps

### ISSUE THREAD

**Issue by teknium1** [tloen/alpaca-lora#391]
We are trying to build a LORA on 30b llama, with latest HF transformers converted model/tokenizer 4.29, latest peft, etc.
We are logging every step

Here is our run params:
```
WORLD_SIZE=1 CUDA_VISIBLE_DEVICES=0,1 python dbg_finetune.py \
    --base_model='/home/jupyter-elinas/models/llama-30b-hf' \
    --data_path='/home/jupyter-elinas/training/alpaca-lora/dataset.json' \
    --num_epochs=3 \
    --learning_rate=0.003 \
    --cutoff_len=1024 \
    --group_by_length \
    --output_dir='./gpt4-llama-30b-it4' \
    --lora_target_modules='[q_proj,k_proj,v_proj,o_proj]' \
    --lora_r=128 \
    --lora_alpha=256 \
    --batch_size=384 \
    --micro_batch_size=12 \
  
```
    Here is our dependency versions:
```
    Package                  Version
------------------------ -----------
accelerate               0.18.0
aiohttp                  3.8.4
aiosignal                1.3.1
appdirs                  1.4.4
asttokens                2.2.1
async-timeout            4.0.2
attrs                    23.1.0
backcall                 0.2.0
bitsandbytes             0.37.2
bitsandbytes-cuda113     0.30.1
black                    23.3.0
certifi                  2022.12.7
charset-normalizer       3.1.0
click                    8.1.3
cmake                    3.26.3
datasets                 2.11.0
decorator                5.1.1
dill                     0.3.6
executing                1.2.0
filelock                 3.12.0
fire                     0.5.0
frozenlist               1.3.3
fsspec                   2023.4.0
huggingface              0.0.1
huggingface-hub          0.13.4
idna                     3.4
ipython                  8.12.0
jedi                     0.18.2
Jinja2                   3.1.2
lit                      16.0.1
loralib                  0.1.1
MarkupSafe               2.1.2
matplotlib-inline        0.1.6
mpmath                   1.3.0
multidict                6.0.4
multiprocess             0.70.14
mypy-extensions          1.0.0
networkx                 3.1
numpy                    1.24.2
nvidia-cublas-cu11       11.10.3.66
nvidia-cuda-cupti-cu11   11.7.101
nvidia-cuda-nvrtc-cu11   11.7.99
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cudnn-cu11        8.5.0.96
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.2.10.91
nvidia-cusolver-cu11     11.4.0.1
nvidia-cusparse-cu11     11.7.4.91
nvidia-nccl-cu11         2.14.3
nvidia-nvtx-cu11         11.7.91
packaging                23.1
pandas                   2.0.0
parso                    0.8.3
pathspec                 0.11.1
peft                     0.3.0.dev0
pexpect                  4.8.0
pickleshare              0.7.5
pip                      21.2.3
platformdirs             3.2.0
prompt-toolkit           3.0.38
protobuf                 3.20.0
psutil                   5.9.5
ptyprocess               0.7.0
pure-eval                0.2.2
pyarrow                  11.0.0
Pygments                 2.15.1
python-dateutil          2.8.2
pytz                     2023.3
PyYAML                   6.0
regex                    2023.3.23
requests                 2.28.2
responses                0.18.0
sentencepiece            0.1.98
setuptools               57.4.0
six                      1.16.0
stack-data               0.6.2
sympy                    1.11.1
termcolor                2.2.0
tokenize-rt              5.0.0
tokenizers               0.13.3
tomli                    2.0.1
torch                    2.0.0
tqdm                     4.65.0
traitlets                5.9.0
transformers             4.29.0.dev0
triton                   2.0.0
typing_extensions        4.5.0
tzdata                   2023.3
urllib3                  1.26.15
wcwidth                  0.2.6
wheel                    0.40.0
xxhash                   3.2.0
yarl                     1.9.1
```

**Comment by teknium1** [comment]
![image](https://user-images.githubusercontent.com/127238744/234468371-4de90083-45c7-46e4-8b2f-2fe429b4b464.png)

**Comment by elven2016** [comment]
I encontered the same issue，have you find the solution?
![image](https://user-images.githubusercontent.com/16677082/234589205-768c6514-bd5b-480a-94ef-fa0a954f8d6c.png)

**Comment by official-elinas** [comment]
Just some debugging

* Setting `--val_set_size=0` does not help
* Deleted HF cached files
* Currently not including `--group_by_length` to see if that makes a difference.

Edit: Removing `--group_by_length` just increased memory usage and eventually went to 0 loss as well.

Other than that, I have no idea what it could be. Also I'm working with @teknium1 on this issue,

**Comment by teknium1** [comment]
It seems it was the LR

**Comment by TingchenFu** [comment]
@teknium1  Hi, I meet with a similar problem in that the loss becomes 0 after several steps. Could you please explain a bit more? How do you solve the problem?

**Comment by teknium1** [comment]
> @teknium1 Hi, I meet with a similar problem in that the loss becomes 0 after several steps. Could you please explain a bit more? How do you solve the problem?

Our learning rate was too high, 0.003 instead of 0.0003
